### PR TITLE
node case insensitive headers

### PR DESCRIPTION
**Pull request by mamaso**
Pulled case-insensitive headers stuff out of #871

### CODE REVIEW

File: src/WebJobs.Script/Content/Script/http/response.js [hunk 1]
```diff
@@ -6,50 +6,73 @@ module.exports = (context) => {
...
             return res;
         },
 
+        // express methods
         status: (statusCode) => {
             res.statusCode = statusCode;
             return res;
         },
 
-        set: (field, val) => {
```
Comment (mamaso) [inline src/WebJobs.Script/Content/Script/http/response.js]: moved below

File: src/WebJobs.Script/Content/Script/http/response.js [hunk 2]
```diff
@@ -6,50 +6,73 @@ module.exports = (context) => {
...
-            return res.set('Content-Type', type);
+            return res.set('content-type', type);
         },
 
         json: (body) => {
             return res.type('application/json')
                 .send(body);
-        },
-
-        raw: (body) => {
```
Comment (mamaso) [inline src/WebJobs.Script/Content/Script/http/response.js]: moved above

File: src/WebJobs.Script/Content/Script/http/response.js [hunk 3]
```diff
@@ -6,50 +6,73 @@ module.exports = (context) => {
...
-            res.isRaw = true;
-            return res.send(body);
-        },
-
-        get: (field) => {
```
Comment (mamaso) [inline src/WebJobs.Script/Content/Script/http/response.js]: moved below

File: WebJobs.Script.proj [hunk 4]
```diff
@@ -178,7 +178,7 @@
     <BuildConfiguration Condition=" $(Configuration) != '' ">$(Configuration)</BuildConfiguration>
   </PropertyGroup>
 
-  <Target Name="MochaTest" DependsOnTargets="Build" AfterTargets="UnitTest">
+  <Target Name="MochaTest" DependsOnTargets="TestBuild" AfterTargets="UnitTest">
```
Comment (mamaso) [inline WebJobs.Script.proj]: Forces a build of the test project, which contains functions.tests.js

File: src/WebJobs.Script/Content/Script/http/response.js [hunk 5]
```diff
@@ -6,50 +6,73 @@ module.exports = (context) => {
...
         body: undefined,
 
+        // functions methods
+        raw: (body) => {
+            res.isRaw = true;
+            return res.send(body);
+        },
+
+        // node httpResponse methods
+        setHeader: (field, val) => {
```
Comment (mathewc) [inline src/WebJobs.Script/Content/Script/http/response.js]: So you're adding set/get/remove for headers, in addition to the existing set method and array access. No objections just wondering what the motivation is for adding more ways to do the same thing?

Comment (mamaso) [inline src/WebJobs.Script/Content/Script/http/response.js]: `setHeader` and the like are methods that exist on node's httpResponse.  Express responses extend httpResponse with methods like `set` and `header`.  All methods are valid from an express point of view, got a customer who was confused at the lack of `setHeader` existence

**Comment by mathewc** [comment]
Just one test comment, otherwise looks good - thanks for splitting this out :) :ship:

File: test/WebJobs.Script.Tests/TestScripts/Node/functions.tests.js [hunk 6]
```diff
@@ -99,9 +128,9 @@ describe('http', () => {
         });
 
         it('get', () => {
-            expect(req.get('test')).to.equal('val');
```
Comment (mamaso) [inline test/WebJobs.Script.Tests/TestScripts/Node/functions.tests.js]: IMO it's not necessary in a simple case like this when uppercase is a superset of lowercase

However let me know if you really want the extra test and I'll add

Comment (mathewc) [inline test/WebJobs.Script.Tests/TestScripts/Node/functions.tests.js]: Yeah, I want a test that explicitly verifies that a header can be retrieved with both upper and lower case names. Easy to do in unit tests.

Comment (mamaso) [inline test/WebJobs.Script.Tests/TestScripts/Node/functions.tests.js]: done

**Pull request merged**